commit 2475c362134a0fa5309c7b0fdb6fc1b86dca88a1
Author: Michael Ellerman <mpe@ellerman.id.au>
Date:   Wed Dec 16 21:26:28 2015 +1100

    Partial revert of "powerpc: Individual System V IPC system calls"
    
    This partially reverts commit a34236155afb1cc41945e58388ac988431bcb0b8.
    
    While reviewing the glibc patch to exploit the individual IPC calls,
    Arnd & Andreas noticed that we were still requiring userspace to pass
    IPC_64 in order to get the new style IPC API.
    
    With a bit of cleanup in the kernel we can drop that requirement, and
    instead only provide the new style API, which will simplify things for
    userspace.
    
    Rather than try and sneak that patch into 4.4, instead we will drop the
    individual IPC calls for powerpc, and merge them again in 4.5 once the
    cleanup patch has gone in.
    
    Because we've already added sys_mlock2() as syscall #378, we don't do a
    full revert of the IPC calls. Instead we drop the __NR #defines, and
    send those now undefined syscall numbers to sys_ni_syscall(). This
    leaves a gap in the syscall numbers, but we'll reuse them when we merge
    the individual IPC calls.
    
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Acked-by: Arnd Bergmann <arnd@arndb.de>

diff --git a/arch/powerpc/include/uapi/asm/unistd.h b/arch/powerpc/include/uapi/asm/unistd.h
index 1effea5193d6..12a05652377a 100644
--- a/arch/powerpc/include/uapi/asm/unistd.h
+++ b/arch/powerpc/include/uapi/asm/unistd.h
@@ -388,18 +388,6 @@
 #define __NR_switch_endian	363
 #define __NR_userfaultfd	364
 #define __NR_membarrier		365
-#define __NR_semop		366
-#define __NR_semget		367
-#define __NR_semctl		368
-#define __NR_semtimedop		369
-#define __NR_msgsnd		370
-#define __NR_msgrcv		371
-#define __NR_msgget		372
-#define __NR_msgctl		373
-#define __NR_shmat		374
-#define __NR_shmdt		375
-#define __NR_shmget		376
-#define __NR_shmctl		377
 #define __NR_mlock2		378
 
 #endif /* _UAPI_ASM_POWERPC_UNISTD_H_ */